### PR TITLE
[FIX] "Speed up" and "Cancel" buttons showing up in the receiver side

### LINKED ISSUE #3222: Txn Status Issue: The speed up and cancel CTA’s appear on the receiver ( the person who received the tokens) end.
**Describe the bug**
The speed up and cancel CTA’s appear on the receiver ( the person who received the tokens) end. Meaning a user sends some eth with a low gas fee to one of his accounts (lets call this the sender account). The sender account then speeds up the transaction. Upon doing so the transaction becomes confirmed. Fine. If you were to go to the receiver account then open up the transactions view, you will see a transaction with the speed up and cancel CTA’s. 

 
**Screenshots**
image

**To Reproduce**
1. Go through the send flow to send some eth to one of your accounts
2. Before submitting your txn change the gas fee to low
3. Submit txn 
4. Go to the transaction overview and speed up the txn
5. Wait until the txn is confirmed then switch to the account that you are receiving funds in
6. go to the transaction overview 
7. notice the bug

**Expected behavior**
The receiver transaction overview should not contain any transaction that requires him/her to speed up the transaction. 

**Smartphone (please complete the following information):**
 - App Version [e.g. 1.0.0] - find version number in app from Settings > About MetaMask

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

The speed up and cancel CTA’s appear on the receiver (the person who received the funds) end.

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #3222

### CODE REVIEW

File: app/util/activity/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		transferInformation,
+	} = tx;
+
+	if (
+		isFromOrToSelectedAddress(from, to, selectedAddress) &&
+		isFromCurrentChain(tx, networkId, chainId) &&
+		tx.status !== TX_UNAPPROVED
+	) {
+		if (isTransfer)
+			return !!tokens.find(({ address }) => toLowerCaseEquals(address, transferInformation.contractAddress));
```
Comment (andreahaku) [inline app/util/activity/index.ts:66]: I'd prefer a ternary here. Something like:
`return isTransfer ? !!tokens.find(({ address }) => toLowerCaseEquals(address, transferInformation.contractAddress)) : true;`

**Review by andreahaku — APPROVED** [review]
LGTM

**Review by plasmacorral — CHANGES_REQUESTED** [review]
Still seeing the issue present in android 11, testing with Moto G play (2021) running December 2021 security updates.

I am able to see the pending transaction from the receiving address side in transaction history.

Device logs:
[bugreport-guamna_t-RZAS31.Q2-146-14-5-2022-02-24-17-58-15.zip](https://github.com/MetaMask/metamask-mobile/files/8137469/bugreport-guamna_t-RZAS31.Q2-146-14-5-2022-02-24-17-58-15.zip)

State Logs:
[PR3790_MetaMask State logs -  v4.1-3.0 (816).zip](https://github.com/MetaMask/metamask-mobile/files/8137474/PR3790_MetaMask.State.logs.-.v4.1-3.0.816.zip)

**Comment by plasmacorral** [comment]
To Reproduce

    1. Go through the send flow to send some eth to one of your accounts (on rinkeby)
    2. Before submitting your txn tap the estimated gas fee
    3. Tap Advanced options.  
    4. Take a few gwei off Max Priority Fee until red network conditions warnings pop 
    5. Do the same with Max Fee, until red network conditions warnings pop
    6. Tap save
    7. Tap Send to submit txn
    8. Tap Back to wallet view
    9. Change to receiving address 
    10. Tap asset (Eth) to see transactions
    11. Note that incoming transaction shows as "Submitted" with speed up and cancel buttons on receiving side in advance of being accepted on chain.

Expected behavior
The receiver transaction overview should not contain any transaction that requires him/her to speed up the transaction.

File: app/util/activity/index.ts (hunk 1 above)
Comment (gantunesr) [reply app/util/activity/index.ts]: Done in [a8d4751](https://github.com/MetaMask/metamask-mobile/pull/3790/commits/a8d475160d80deaf77750438e7673248efc47b9d)

**Comment by gantunesr** [comment]
Fixed in [b924dcc](https://github.com/MetaMask/metamask-mobile/pull/3790/commits/b924dccdfb847f44e3326601260b193657e05e40) and [ccf6d5f](https://github.com/MetaMask/metamask-mobile/pull/3790/commits/ccf6d5fc8010d37679ae622d66a9c75efe6a5e8a)

cc @plasmacorral

**Review by cortisiko — CHANGES_REQUESTED** [review]
@gantunesr sending this back your way:  The sender’s pending transaction does not appear in the activity view but appears on asset overview. Meaning Once I send a transaction between my wallet address with a low gas fee, the pending transaction does not appear in the activity view (formerly transaction history). 
 See recording [here](http://recordit.co/ChwunZ5nRS) 

_**The transaction appeared after it was confirmed.**_ 

to reproduce:

Go through the send flow to send some eth to one of your accounts
Before submitting your txn change the gas fee to low
Submit txn
go to the activity view ( formerly transaction history) 
notice the pending transaction is missing.

**Comment by gantunesr** [comment]
@cortisiko fixed in [ad4b9eb](https://github.com/MetaMask/metamask-mobile/pull/3790/commits/ad4b9eb81fd3bc3e0f4eb4d0013ed2aa7e7d4efb)

**Review by cortisiko — APPROVED** [review]
I tested:
sending ETH and speeding up the transaction
                                  -  between wallet addresses
                                  - between different wallets

sending ERC-20 tokens and speeding up the transaction
                                  -  between wallet addresses
                                  - between different wallets

sending an NFT and speeding up the transaction:
                                  -  between wallet addresses
                                  - between different wallets

I verified that the asset overview and transaction activity on both the sender and receiver showed the correct state. I also verified that the receiver of the transaction did not see duplicate transactions and the pending transaction did not show up on the receivers end. 

@gantunesr this guy is 🌮  🌮 🌮

**Review by plasmacorral — APPROVED** [review]
Approving changes based on cortisiko's testing

**Pull request merged**